### PR TITLE
Adds gzstream stream, exposes to swig

### PR DESCRIPTION
**Pull request by bp-kelley**
Currently only tested in Java.  Usage:

```
          gzstream stream = new gzstream(fileN.getPath());
          ForwardSDMolSupplier suppl = new ForwardSDMolSupplier(stream);
```

TODO:  Add gzstream for writers.
TODO:  Add ForwardSmilesMolSupplier (might think about a more modern csv parser)

### CODE REVIEW

**Comment by bp-kelley** [comment]
Wonderful, fails on windows...

**Comment by greglandrum** [comment]
A thought that just occurred to me: 

This creates a new class and some complication and, in the end, just enables gzipped files to be read. Wouldn't this be a ton simpler and at least 90% as useful if we changed `ForwardMolSupplier` to recognize the `.gz` extension and kept all the iostreams stuff on the C++ side?
This would let me do either `ForwardSDMolSupplier suppl = new ForwardSDMolSupplier("foo.sdf");` or `ForwardSDMolSupplier suppl = new ForwardSDMolSupplier("foo.sdf.gz");`

**Comment by greglandrum** [comment]
(sorry, accidentally hit "close and comment" instead of "comment")

**Comment by bp-kelley** [comment]
@greglandrum I thought about that, but it adds a non header dependency to just about everything.  I think this is cleaner.

**Comment by greglandrum** [comment]
How would the dependency situation be different from what this requires?
In each case we'd need to link boost::iostreams wherever we use FileParsers, right?

**Comment by greglandrum** [comment]
Never mind, answered my own question.

**Comment by greglandrum** [comment]
@bp-kelley : I'm not sure if this builds once merged. Do you want me to fix the conflicts?

**Comment by bp-kelley** [comment]
I’m checking on windows now and will update the pr

**Comment by bp-kelley** [comment]
Windows is working now, as long as you don't use boost 1.62 :)

**Comment by bp-kelley** [comment]
@greglandrum  ready for final review

File: Code/GraphMol/FileParsers/CMakeLists.txt [hunk 1]
```diff
@@ -40,20 +40,24 @@ rdkit_test(fileParsersTest1 test1.cpp
            LINK_LIBRARIES FileParsers SmilesParse
 	   SubstructMatch GraphMol RDGeneral RDGeometryLib ${maeparser_var})
 
-if(RDK_BUILD_TEST_GZIP)
+
```
Comment (greglandrum) [inline Code/GraphMol/FileParsers/CMakeLists.txt]: there's some duplication here. I think you meant to delete this block

File: Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SuppliersTests.java [hunk 2]
```diff
@@ -167,6 +167,28 @@ public void test10Leak() {
...
             }
 	}
+  
+        @Test
+        public void test11GZstream() {
+          // NCI_aids_few.sdf.gz
+          File base = getRdBase();
+          File gzpath  = new File(base, "Code" + File.separator + "GraphMol" + File.separator +
+                                  "FileParsers" + File.separator + "test_data");
+          File fileN = new File(baseTestPath, "NCI_aids_few.sdf.gz");
```
Comment (greglandrum) [inline Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SuppliersTests.java]: looks like you forgot to add this file

Comment (bp-kelley) [reply Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SuppliersTests.java]: I'm using the file in the FileParsers suite:

https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/FileParsers/test_data/NCI_aids_few.sdf.gz

File: Code/GraphMol/FileParsers/CMakeLists.txt (hunk 1 above)
Comment (bp-kelley) [reply Code/GraphMol/FileParsers/CMakeLists.txt]: Will do.

**Comment by bp-kelley** [comment]
It looks like boost 1.65.0 is failing on windows (hopefully).  I'll check into updating appveyor

File: Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SuppliersTests.java (hunk 2 above)
Comment (greglandrum) [reply Code/JavaWrappers/gmwrapper/src-test/org/RDKit/SuppliersTests.java]: ah, sorry. I looked in the wrong place. Reviewer-error.

**Comment by bp-kelley** [comment]
@greglandrum This should fix the FileParsers static issues.

**Comment by greglandrum** [comment]
This builds for me on Windows as well. Merging

**Pull request merged**